hadoop-ozone/ozone-recon/src/main/java/org/apache/hadoop/ozone/recon/spi/impl/ContainerDBServiceProviderImpl.java

```diff
@@ -271,8 +332,54 @@ public void deleteContainerMapping(ContainerKeyPrefix containerKeyPrefix)
     containerKeyTable.delete(containerKeyPrefix);
   }
 
+  /**
+   * Get total count of containers.
+   *
+   * @return total count of containers.
+   */
+  @Override
+  public long getCountForContainers() {
+    GlobalStats containerCountRecord =
+        globalStatsDao.fetchOneByKey(CONTAINER_COUNT_KEY);
+
+    return (containerCountRecord == null) ? 0L :
+        containerCountRecord.getValue();
+  }
+
   @Override
   public TableIterator getContainerTableIterator() {
     return containerKeyTable.iterator();
   }
+
+  /**
+   * Store the total count of containers into the container DB store.
+   *
+   * @param count count of the containers present in the system.
+   */
+  @Override
+  public void storeContainerCount(Long count) {
+    // Get the current timestamp
+    Timestamp now =
```

Comment:
Any reason to ask sql timestamp vs System.currentMillis()?

Reply:
Yes. The assumption here is that the SQL database will be running in UTC timezone as a standard practice for all production databases and we don't want to mess with the timezone by using System.currentMillis().

Reply:
But Ozone will rely on JVM timestamp, and since time is relative, as long as we don't change the impl midway this point is moot, isn't it :-)

Reply:
At some point in the future, we might show the last updated timestamp in the UI and that is why using SQL timestamp here. As long as this value is only used by the application, we can use System.currentMillis() and that is what we are doing for lastUpdatedTimestamp in ReconInternalSchema.